darwin/utils.py

```diff
@@ -22,3 +55,93 @@ def prompt(msg: str, default: Optional[str] = None) -> str:
     if not result and default:
         return default
     return result
+
+
+def find_files(
+    root: Optional[Path] = None,
+    files_list: Optional[List[str]] = None,
+    recursive: bool = True,
+    exclude: Optional[List[str]] = None,
+) -> List[Path]:
+    """Retrieve a list of all files belonging to supported extensions. The exploration can be made
+    recursive and a list of files can be excluded if desired.
+
+    Parameters
+    ----------
+    root : Path
+        Path to the root folder to explore
+    recursive : bool
+        Flag for recursive search
+    exclude : list[str]
+        List of files to exclude from the search
+
+    Returns
+    -------
+    list[Path]
+    List of all files belonging to supported extensions
+    """
+    if exclude is None:
+        exclude = []
+    if files_list is not None:
+        files: List[Path] = []
+        for file in files_list:
+            file = Path(file)
+            if (
+                file.suffix in SUPPORTED_IMAGE_EXTENSIONS + SUPPORTED_VIDEO_EXTENSIONS
+                and file.suffix not in exclude
+            ):
+                files += [file]
+        return files
+    if root is not None:
+        if not root.is_dir():
+            # print ("TODO: when are we supposed to enter here")
+            if (
+                root.suffix in SUPPORTED_IMAGE_EXTENSIONS + SUPPORTED_VIDEO_EXTENSIONS
+                and root.suffix not in exclude
+            ):
+                return [root]
+            else:
+                return []
+        files: List[Path] = []
+        for file in root.iterdir():
+            if file.is_dir():
+                if recursive:
+                    files += find_files(root=file, recursive=recursive, exclude=exclude)
+            else:
+                if (
+                    file.suffix in SUPPORTED_IMAGE_EXTENSIONS + SUPPORTED_VIDEO_EXTENSIONS
+                    and file.suffix not in exclude
+                ):
+                    files += [file]
+        return files
+
+
+def secure_continue_request():
+    """Asks for explicit approval from the user. Empty string not accepted"""
+    return input("Do you want to continue? [y/N] ") in ["Y", "y"]
+
+
+def make_configuration_file(client: "Client") -> Config:
```

Comment:
I would probably rename it and take the config_path as an optional parameter
persist_client_configuration(client: "Client", config_path: Optional[Path] = None) -> Config

Reply:
Done